ipc/uapi_illumos.go

```diff
@@ -0,0 +1,146 @@
+// +build illumos
+
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2017-2019 WireGuard LLC. All Rights Reserved.
+ * Copyright 2019 Joyent, Inc.
+ */
+
+package ipc
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"net"
+	"os"
+	"path"
+)
+
+var socketDirectory = "/var/run/wireguard"
+
+const (
+	IpcErrorIO        = -int64(unix.EIO)
+	IpcErrorProtocol  = -int64(unix.EPROTO)
+	IpcErrorInvalid   = -int64(unix.EINVAL)
+	IpcErrorPortInUse = -int64(unix.EADDRINUSE)
+	socketName        = "%s.sock"
+)
+
+type UAPIListener struct {
+	listener net.Listener // unix socket listener
+	connNew  chan net.Conn
```

Comment:
Oh, you were wondering about SMF to manage wireguard-go's lifetime, not wg-quick's. Gotcha. Same response here, though -- let's roll with the usual wireguard-go semantics and not try to do too much that's fancy right away.

Reply:
Okay, so that means just backgrounding wireguard-go, same as the other systems. Easy enough. @nshalman in that case, you should adapt the BSD code's use of kqueue to Illumos' file event ports. Unfortunately, that's more API surface that isn't exposed in x/sys/unix. We'd want to take inspiration from the PR at https://github.com/fsnotify/fsnotify/pull/263/files to get the right port primitives into x/sys/unix as well.

Reply:
@nshalman - I too am wondering about where the file watcher stuff is for this. UAPI on other platforms is able to notify when the socket has been removed.

Reply:
I've got open CRs to get the event ports APIs from the fsnotify PR into x/sys/unix and a fork of fsnotify that uses them and passes all integration tests. Would it make sense to use fsnotify or is there a preference to use event ports directly?

Reply:
https://github.com/fsnotify/fsnotify/pull/371 is still a draft, but the underlying event ports support is upstream in x/sys/unix now.

Reply:
Event Ports is in x/sys/unix and being used here.